frontend/src/main/scala/bloop/exec/JvmProcessForker.scala

```diff
@@ -113,23 +117,65 @@ final class JvmForker(config: JdkConfig, classpath: Array[AbsolutePath]) extends
       extraClasspath: Array[AbsolutePath]
   ): Task[Int] = {
     val jvmOptions = jargs.map(_.stripPrefix("-J")) ++ config.javaOptions
-    val fullClasspath = (classpath ++ extraClasspath).map(_.syntax).mkString(pathSeparator)
-    Task.fromTry(javaExecutable).flatMap { java =>
-      val classpathOption = "-cp" :: fullClasspath :: Nil
-      val appOptions = mainClass :: args.toList
-      val cmd = java.syntax :: jvmOptions.toList ::: classpathOption ::: appOptions
-      val logTask = if (logger.isVerbose) {
+    val fullClasspath = classpath ++ extraClasspath
+
+    for {
+      fullClasspathStr <- Task.fromTry(ensureClasspathLength(fullClasspath))
+      java <- Task.fromTry(javaExecutable)
+
+      classpathOption = "-cp" :: fullClasspathStr :: Nil
+      appOptions = mainClass :: args.toList
+      cmd = java.syntax :: jvmOptions.toList ::: classpathOption ::: appOptions
+
+      logTask <- if (logger.isVerbose) {
         val debugOptions =
           s"""
              |Fork options:
              |   command      = '${cmd.mkString(" ")}'
              |   cwd          = '$cwd'""".stripMargin
         Task(logger.debug(debugOptions)(DebugFilter.All))
-      } else Task.unit
-      logTask.flatMap(_ => Forker.run(cwd, cmd, logger, opts))
+      } else {
+        Task.unit
+      }
+      res <- Forker.run(cwd, cmd, logger, opts)
+    } yield {
+      res
     }
   }
 
+  private def ensureClasspathLength(classpath: Array[AbsolutePath]): Try[String] = {
+    val fullClasspathStr = classpath.map(_.syntax).mkString(pathSeparator)
+
+    if (fullClasspathStr.length > JvmProcessForker.classpathCharLimit)
+      createTempManifestJar(classpath).map(_.syntax)
+    else
+      Try(fullClasspathStr)
+  }
+
+  private def createTempManifestJar(classpath: Array[AbsolutePath]): Try[AbsolutePath] = Try {
+    val prefix = "jvm-forker-manifest"
+    val manifestJar = Files.createTempFile(prefix, ".jar").toAbsolutePath
```

Comment:
Should we not be comparing the chars of the full cmd invocation instead of the classpath? We could have a classpath that doesn't violate the chars limit but that combined with the other jvm options does. In that case, we would not create a MANIFEST but we should have, as the underlying OS will fail.

Reply:
will look at full cmd, but descoping to just classpath shortening